### PR TITLE
Add support for efficient range checking to RangeSet

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: src/core/rangeset.js [hunk 1]
```diff
@@ -1,67 +1,61 @@
...
-  ledger = Number(ledger);
+  addValue(value: number) {
+    this.addRange(value, value);
+  }
 
-  return this._ranges.some(function(r) {
-    return ledger >= r.start && ledger <= r.end;
-  });
-};
+  addParsedRanges(rangesString: string) {
```
Comment (wltsmrz) [inline src/core/rangeset.js]: You're really adding unparsed ledger ranges, right? Maybe `addRanges` is good enough

File: test/rangeset-test.js [hunk 2]
```diff
@@ -1,78 +1,49 @@
...
-      r.add(void(0));
-    });
-    assert.throws(function() {
-      r.add('a');
-    });
+  it('addValue()/addRange() -- malformed', function() {
+    const r = new RangeSet();
     assert.throws(function() {
-      r.add('2-1');
+      r.addRange(2, 1);
```
Comment (wltsmrz) [inline test/rangeset-test.js]: How about a test for `addParsedRanges`?

File: src/core/rangeset.js (hunk 1 above)
Comment (clark800) [inline src/core/rangeset.js]: An unparsed range would be a string, a parsed range is a range. The other name I considered was `parseRangesAndAdd`, but I thought `addParsedRanges` sounded simpler.

Comment (clark800) [inline src/core/rangeset.js]: I can see that it's ambiguous though. Is `parseRangesAndAdd` better?

Comment (wltsmrz) [inline src/core/rangeset.js]: I'd vote for either addRanges or addSerializedRanges; addParsedRanges seems confusing

**Comment by wltsmrz** [comment]
LGTM otherwise

File: src/core/rangeset.js (hunk 1 above)
Comment (clark800) [inline src/core/rangeset.js]: I went with `parseAndAddRanges` since it's the least ambiguous.

**Pull request merged**